src/main/java/baritone/utils/ToolSet.java

```diff
@@ -23,87 +23,140 @@
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.init.Enchantments;
 import net.minecraft.init.MobEffects;
+import net.minecraft.item.Item.ToolMaterial;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemTool;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Function;
 
 /**
  * A cached list of the best tools on the hotbar for any block
  *
- * @author avecowa, Brady, leijurv
+ * @author Avery, Brady, leijurv
  */
 public class ToolSet implements Helper {
 
     /**
      * A cache mapping a {@link Block} to how long it will take to break
      * with this toolset, given the optimum tool is used.
      */
-    private Map<Block, Double> breakStrengthCache = new HashMap<>();
+    private final Map<Block, Double> breakStrengthCache = new HashMap<>();
+
+    /**
+     * My buddy leijurv owned me so we have this to not create a new lambda instance.
+     */
+    private final Function<Block, Double> backendCalculation = this::getBestDestructionTime;
+
+    /**
+     * Using the best tool on the hotbar, how long would it take to mine this block
+     *
+     * @param state the blockstate to be mined
+     * @return how long it would take in ticks
+     */
+    public double getStrVsBlock(IBlockState state) {
+        return breakStrengthCache.computeIfAbsent(state.getBlock(), backendCalculation) * potionAmplifier();
```

Comment:
Also, I'm not 100% sure about having potion effects here like this. I think it's good that it's it's own function but I think it can be cached, no? What's the point of pulling it out of the cache? Like, if you have haste at the start of path calculation but it runs out within the 500ms before the timeout? This toolset object is used only for one calculation, because we can't assume that the tools they have on their hotbar (or their potion effects) haven't changed from the last path calculation

Reply:
The cache is persistant across paths. Each segment should use updated potion statuses. I'll work on a better way of making this work. Ideally it would know how long it would take to get to a node and whether the effect is still active by then, but that's a bit over complicated. However storing the result for block values persistantly is an issue across segments. Each segment should recalculate the toolset map.